src/config/config.cc

```diff
@@ -658,7 +658,13 @@ Status Config::parseConfigFromPair(const std::pair<std::string, std::string> &in
     field_key = input.first;
     tokens[input.second] = input.first.substr(ns_str_size);
   }
+
   auto iter = fields_.find(field_key);
+
+  if (iter == fields_.end()) {
```

Comment:
Would you like to move the warning message inside `else` branch of `if (iter != fields_.end()) {` check which is located later? 
P.S. I think you can omit a lot of `!` inside the message, just capitalize the word 'warning' as `WARNING:...`

Reply:
Sure can